pkgs/top-level/java-packages.nix

```diff
@@ -7,27 +7,46 @@ with pkgs;
 
   compiler =
     let
+      mergeMetaPlatforms =
+        jdk: other:
+        jdk
+        // {
+          meta = jdk.meta // {
+            platforms = lib.unique (jdk.meta.platforms ++ other.meta.platforms);
+          };
+        };
+
       mkOpenjdk =
         featureVersion: path-darwin:
-        if stdenv.hostPlatform.isLinux then
-          mkOpenjdkLinuxOnly featureVersion
-        else
-          let
-            openjdk = callPackage path-darwin { };
-          in
-          openjdk // { headless = openjdk; };
-
-      mkOpenjdkLinuxOnly =
-        featureVersion:
         let
-          openjdk = callPackage ../development/compilers/openjdk/generic.nix { inherit featureVersion; };
+          # merge meta.platforms of both packages so that dependent packages and hydra build them
+          mergeMetaPlatforms =
+            jdk: other:
+            jdk
+            // {
+              meta = jdk.meta // {
+                platforms = lib.unique (jdk.meta.platforms ++ other.meta.platforms);
+              };
+            };
```

Comment:
```suggestion
```

Isn't this already in scope from the top of the file?

Reply:
Some hikup while doing the 5th rebase